src/openloco/windowmgr.cpp

```diff
@@ -136,4 +138,60 @@ namespace openloco::ui::windowmgr
         regs.esi = (uint32_t)window;
         call(0x4ca17f, regs);
     }
+
+    // 0x004CD3D0
+    void dispatch_update_all()
+    {
+        _523508++;
+        companymgr::updating_company_id(companymgr::get_controlling_id());
+
+        for (ui::window* w = _windows_end - 1; w >= _windows; w--)
+        {
+            w->call_update();
+        }
+
+        call(0x4CE6FF);
+        call(0x4CEEA7);
+    }
+
+    // 0x004CC6EA
+    void close(window* window)
+    {
+        if (window == nullptr)
+        {
+            return;
+        }
+
+        // Make a copy of the window class and number in case
+        // the window order is changed by the close event.
+        auto type = (window_type)window->type;
+        uint16_t number = window->number;
+
+        window->call_close();
+
+        window = find(type, number);
+        if (window == nullptr)
+            return;
+
+        if (window->viewports[0] != nullptr)
+        {
+            window->viewports[0]->width = 0;
+            window->viewports[0] = nullptr;
+        }
+
+        if (window->viewports[1] != nullptr)
+        {
+            window->viewports[1]->width = 0;
+            window->viewports[1] = nullptr;
+        }
+
+        window->invalidate();
+        _windows_end--;
+        auto windowCount = ((uintptr_t)*_windows_end - (uintptr_t)window) / sizeof(ui::window);
```

Comment:
Is there any reason to change these into uintptr_t's cant you just perform the pointer maths on them then you dont need to divide by sizeof.

Reply:
For me, the `loco_global<>` pointers are weird. CLion doesn't give really usable autocompletion, so I just resorted to copying what was in the `num_windows()` function. I'm not even sure if it's using the correct pointers at this point.